regress/unittests/win32compat/miscellaneous_tests.c

```diff
@@ -317,100 +317,101 @@ test_chroot()
-	buf[len_pg] = '\0';
-	strcat(buf, "\\sftp-server.exe\" -arg\"");
-	ASSERT_STRING_EQ(out, buf);
-	out = build_session_commandline("c:\\system32\\cmd.exe", NULL, "sftp-SERVER -arg");
-	buf[len_pg] = '\0';
-	strcat(buf, "\\sftp-server.exe\" -arg\"");
-	ASSERT_STRING_EQ(out, buf);
-	out = build_session_commandline("c:\\system32\\cmd.exe", NULL, "sCp -arg");
-	buf[len_pg] = '\0';
-	strcat(buf, "\\scp.exe\" -arg\"");
-	ASSERT_STRING_EQ(out, buf);
-	free(out);
-	TEST_DONE();
-	
-	TEST_START("bash shell tests");
-	out = build_session_commandline("c:\\system32\\bash.exe", NULL, "internal-sftp -arg");
-	ASSERT_STRING_EQ(out, "\"c:\\system32\\bash.exe\" -c \"sftp-server.exe -arg\"");
+	TEST_START("scp tests");
+	out = build_exec_command("sCp -arg");	
+	ASSERT_STRING_EQ(out, "scp.exe -arg");
 	free(out);
-	out = build_session_commandline("c:\\system32\\bash", NULL, "internal-sftp -arg");
-	ASSERT_STRING_EQ(out, "\"c:\\system32\\bash\" -c \"sftp-server.exe -arg\"");
+	out = build_exec_command("sCp.exe -arg1 -arg2");
+	ASSERT_STRING_EQ(out, "scp.exe -arg1 -arg2");
 	free(out);
-	out = build_session_commandline("c:\\system32\\bash", NULL, "sFTP-server -arg");
-	ASSERT_STRING_EQ(out, "\"c:\\system32\\bash\" -c \"sftp-server.exe -arg\"");
-	free(out);
-	out = build_session_commandline("c:\\system32\\bash", NULL, "scP -arg");
-	ASSERT_STRING_EQ(out, "\"c:\\system32\\bash\" -c \"scp.exe -arg\"");
-	free(out);
-	out = build_session_commandline("c:\\cygwin\\bash.exe", NULL, "internal-sftp -arg");
-	ASSERT_STRING_EQ(out, "\"c:\\cygwin\\bash.exe\" -c \"sftp-server.exe -arg\"");
-	free(out);
-	out = build_session_commandline("c:\\cygwin\\bash", NULL, "sftp-server -arg");
-	ASSERT_STRING_EQ(out, "\"c:\\cygwin\\bash\" -c \"sftp-server.exe -arg\"");
+	TEST_DONE();
+
+	TEST_START("sftp tests");
+	out = build_exec_command("internal-sftp \"arg1 arg2\"");
+	ASSERT_STRING_EQ(out, "sftp-server.exe \"arg1 arg2\"");
 	free(out);
-	out = build_session_commandline("c:\\cygwin\\bash", NULL, "sftp-seRVer.exe -arg");
-	ASSERT_STRING_EQ(out, "\"c:\\cygwin\\bash\" -c \"sftp-server.exe -arg\"");
+	out = build_exec_command("SFTP-server.exe -arg");
+	ASSERT_STRING_EQ(out, "sftp-server.exe -arg");
 	free(out);
-	out = build_session_commandline("c:\\cygwin\\bash", NULL, "sCp -arg");
-	ASSERT_STRING_EQ(out, "\"c:\\cygwin\\bash\" -c \"scp.exe -arg\"");
+	out = build_exec_command("sftp-SERVER -arg");	
+	ASSERT_STRING_EQ(out, "sftp-server.exe -arg");
 	free(out);
 	TEST_DONE();
+}
 
-	TEST_START("powershell shell tests");
-	out = build_session_commandline("c:\\powershell.exe", NULL, "internal-sftp -arg");
-	ASSERT_STRING_EQ(out, "\"c:\\powershell.exe\" -c \"sftp-server.exe -arg\"");
-	free(out);
-	out = build_session_commandline("c:\\powershell", NULL, "sftp-server -arg");
-	ASSERT_STRING_EQ(out, "\"c:\\powershell\" -c \"sftp-server.exe -arg\"");
+char *
+build_commandline_string(const char* cmd, char *const argv[], BOOLEAN prepend_module_path);
```

Comment:
you can get rid of this by moving it to misc_internal

Reply:
sure